dali/operators/imgcodec/decoder.cc

```diff
@@ -0,0 +1,411 @@
+      0)
+  .AddOptionalArg("hw_decoder_load",
+      R"code(The percentage of the image data to be processed by the HW JPEG decoder.
+
+Applies **only** to the ``mixed`` backend type in NVIDIA Ampere GPU architecture.
+
+Determines the percentage of the workload that will be offloaded to the hardware decoder,
+if available. The optimal workload depends on the number of threads that are provided to
+the DALI pipeline and should be found empirically. More details can be found at
+https://developer.nvidia.com/blog/loading-data-fast-with-dali-and-new-jpeg-decoder-in-a100)code",
+      0.65f)
+  .AddOptionalArg("preallocate_width_hint",
+      R"code(Image width hint.
+
+Applies **only** to the ``mixed`` backend type in NVIDIA Ampere GPU architecture.
+
+The hint is used to preallocate memory for the HW JPEG decoder.)code",
+      0)
+  .AddOptionalArg("preallocate_height_hint",
+      R"code(Image width hint.
+
+Applies **only** to the ``mixed`` backend type in NVIDIA Ampere GPU architecture.
+
+The hint is used to preallocate memory for the HW JPEG decoder.)code",
+      0)
+  .AddOptionalArg("affine",
+      R"code(Applies **only** to the ``mixed`` backend type.
+
+If set to True, each thread in the internal thread pool will be tied to a specific CPU core.
+Otherwise, the threads can be reassigned to any CPU core by the operating system.)code",
+      true)
+  .AddOptionalArg("split_stages",
+      R"code(Applies **only** to the ``mixed`` backend type.
+
+If True, the operator will be split into two sub-stages: a CPU and GPU one.)code",
+      false)
+  .DeprecateArg("split_stages")  // deprecated in DALI 1.0
+  .AddOptionalArg("use_chunk_allocator",
+      R"code(**Experimental**, applies **only** to the ``mixed`` backend type.
+
+Uses the chunk pinned memory allocator and allocates a chunk of the
+``batch_size * prefetch_queue_depth`` size during the construction and suballocates
+them at runtime.)code",
+      false)
+  .DeprecateArg("use_chunk_allocator")  // deprecated in DALI 1.0
+  .AddOptionalArg("use_fast_idct",
+      R"code(Enables fast IDCT in the libjpeg-turbo based CPU decoder, used when ``device`` is set
+to "cpu" or when the it is set to "mixed" but the particular image can not be handled by
+the GPU implementation.
+
+According to the libjpeg-turbo documentation, decompression performance is improved by up to 14%
+with little reduction in quality.)code",
+      false)
+  .AddOptionalArg("memory_stats",
+      R"code(Applies **only** to the ``mixed`` backend type.
+
+Prints debug information about nvJPEG allocations. The information about the largest
+allocation might be useful to determine suitable values for ``device_memory_padding`` and
+``host_memory_padding`` for a dataset.
+
+.. note::
+  The statistics are global for the entire process, not per operator instance, and include
+  the allocations made during construction if the padding hints are non-zero.
+)code",
+      false)
```

Comment:
I'd gladly drop this particular feature - we should rather get rid of this special memory pool. I think this new operator is a good place to cut this off.

Reply:
Marked as deprecated